### PR TITLE
Upgrade `databricks-sdk-go` from 0.7.0 to 0.8.0

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes

Refactoring following the changes in databricks-sdk-go 0.8.0 [release](https://github.com/databricks/databricks-sdk-go/releases/tag/v0.8.0)

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

File: mws/mws.go [hunk 1]
```diff
@@ -72,7 +72,7 @@ type Network struct {
 	GcpNetworkInfo   *GcpNetworkInfo      `json:"gcp_network_info,omitempty"`
 }
 
-// GcpVpcEndpointInfo is the objecy that configures GCP Private Service Connect endpoints.
+// GcpVpcEndpointInfo is the objecy that configures GCP Private Service Connect serving.
```
Comment (alexott) [inline mws/mws.go:75]: is it really `serving`, not `endpoints`?

File: pipelines/resource_pipeline_test.go [hunk 2]
```diff
@@ -474,7 +474,7 @@ func TestResourcePipelineDelete(t *testing.T) {
 		Fixtures: []qa.HTTPFixture{
 			{
 				Method:   "DELETE",
-				Resource: "/api/2.0/pipelines/abcd",
+				Resource: "/api/2.0/pipelines/abcd?",
```
Comment (alexott) [inline pipelines/resource_pipeline_test.go:477]: I'm not sure that we need to add `?` if query string is empty. 
I would say that this is a Go SDK problem

**Review by alexott — APPROVED** [review]
in general looks good, just some comments...

File: mws/mws.go (hunk 1 above)
Comment (nkvuong) [reply mws/mws.go]: pitfall of bulk replacing

**Review by nfx — APPROVED** [review]
thank you for doing this!

**Pull request merged**